server/src/main/scala/com/cloudera/livy/server/LivyServer.scala

```diff
@@ -275,6 +277,17 @@ class LivyServer extends Logging {
       case _ => false
     }
     require(supportedVersion, s"Unsupported Spark version $version.")
+
+    // Spark >= 1.6 require for Livy yarn-master mode
+    if (sparkMaster == "yarn-master") {
```

Comment:
I'm pretty sure this can be just "yarn" (and have `LIVY_SPARK_DEPLOY_MODE` set to "cluster"). So maybe just check for `startsWith` instead of equality.

Also there's no such thing as "yarn-master".